contracts/truefi2/CrvBaseRateOracle.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.10;
+
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+import "../truefi/interface/ICurve.sol";
+
+// prettier-ignore
+contract CrvBaseRateOracle {
+    using SafeMath for uint256;
+    using SafeMath for uint8;
+
+    ICurve public curve;
+
+    // A cyclic buffer structure for storing historical values.
+    // insertIndex points to the place where the next value
+    // should be inserted.
+    struct HistoricalRatesBuffer {
+        uint256[BUFFER_SIZE] baseRates;
+        uint256[BUFFER_SIZE] timestamps;
+        uint8 insertIndex;
+    }
+    HistoricalRatesBuffer public histBuffer;
+
+    // A fixed amount of time to wait
+    // to be able to update the historical buffer
+    uint256 public cooldownTime;
+
+    uint8 public constant BUFFER_SIZE = 7;
+
+    /**
+     * @dev Throws if cooldown is on when updating the historical buffer
+     */
+    modifier offCooldown() {
+        uint256 lastUpdated = histBuffer.timestamps[histBuffer.insertIndex.add(BUFFER_SIZE).sub(1) % BUFFER_SIZE];
+        require(now >= lastUpdated.add(cooldownTime), "CrvBaseRateOracle: Buffer on cooldown");
+        _;
+    }
+
+    constructor(ICurve _curve) public {
+        curve = _curve;
+        cooldownTime = 1 days;
```

Comment:
I feel like this makes more sense as an _argument_ to the constructor rather than a hardcoded constant? works either way though